controllers/users_test.go

```diff
@@ -46,40 +52,157 @@ func createUserTestData() (*test.OrderedTests, error) {
 		AssetsID:   assetID,
 	}
 
-	testCase := "no_existing_user"
-	data := test.Data{
-		Data:     make(map[string]interface{}),
-		Expected: make(map[string]interface{}),
+	nomineeID, err := uuid.NewUUID()
+	if err != nil {
+		return nil, err
 	}
 
-	data.Data.(map[string]interface{})["input"] = user
-	data.Data.(map[string]interface{})["db_mock"] = nil
-	data.Expected.(map[string]interface{})["data"] = &user
-	data.Expected.(map[string]interface{})["error"] = nil
-	dataSet.TestDataSet[testCase] = data
-	dataSet.OrderedList = append(dataSet.OrderedList, testCase)
-
-	testCase = "existing_user"
-	data = test.Data{
-		Data:     make(map[string]interface{}),
-		Expected: make(map[string]interface{}),
+	productID, err := uuid.NewUUID()
+	if err != nil {
+		return nil, err
+	}
+
+	usersProducts := models.UserProducts{
+		ProductMap: make(map[uuid.UUID]int),
 	}
 
-	data.Data.(map[string]interface{})["input"] = user
-	data.Data.(map[string]interface{})["db_mock"] = &user
-	data.Expected.(map[string]interface{})["data"] = nil
-	data.Expected.(map[string]interface{})["error"] = ErrDuplicateEmailEntry
-	dataSet.TestDataSet[testCase] = data
-	dataSet.OrderedList = append(dataSet.OrderedList, testCase)
+	product := models.Product{
+		ID:     productID,
+		Public: true,
+	}
+
+	privileges := make(models.Privileges, 2)
+	privileges[0].ID = 0
+	privileges[0].Name = "Owner"
+	privileges[0].Description = "description0"
+	privileges[1].ID = 1
+	privileges[1].Name = "User"
+	privileges[1].Description = "description1"
+
+	switch testID {
+	case CreateUserTest:
+		testCase := "no_existing_user"
+		data := test.Data{
+			Data:     make(map[string]interface{}),
+			Expected: make(map[string]interface{}),
+		}
+
+		data.Data.(map[string]interface{})["input"] = user
+		data.Data.(map[string]interface{})["db_mock"] = nil
+		data.Expected.(map[string]interface{})["data"] = &user
+		data.Expected.(map[string]interface{})["error"] = nil
+		dataSet.TestDataSet[testCase] = data
+		dataSet.OrderedList = append(dataSet.OrderedList, testCase)
+
+		testCase = "existing_user"
+		data = test.Data{
+			Data:     make(map[string]interface{}),
+			Expected: make(map[string]interface{}),
+		}
+
+		data.Data.(map[string]interface{})["input"] = user
+		data.Data.(map[string]interface{})["db_mock"] = &user
+		data.Expected.(map[string]interface{})["data"] = nil
+		data.Expected.(map[string]interface{})["error"] = ErrDuplicateEmailEntry
+		dataSet.TestDataSet[testCase] = data
+		dataSet.OrderedList = append(dataSet.OrderedList, testCase)
+	case DeleteUserTest:
+
+		testCase := "valid_data_has_nominee"
+		data := test.Data{
+			Data:     make(map[string]interface{}),
+			Expected: make(map[string]interface{}),
+		}
+
+		usersProducts.ProductMap[productID] = 0
+		nominees := make(map[uuid.UUID]uuid.UUID)
+		nominees[productID] = nomineeID
+		data.Data.(map[string]interface{})["user_id"] = user.ID
```

Comment:
Why don't you simply use objects?

Reply:
Can you explain what you mean?

Reply:
Why don't you create a struct to hold these values?
```
type Product struct {
  ProductID uuid.UUID
  Nominees [] uuid.UUUD
 ... 
}
```

Reply:
Potentially each test, could require its own structure. I just feel using a map gives more flexibility. I started using pytest for creating functional tests it gave me some minor optimization ideas. I may raise a ticket and try to create test data generation in a somewhat similar way in the future.

Reply:
I think, this "map thing" just makes harder to read the test code. If each test method or at least, each test file has its own structure, the readibility could be increased.